app/models/user.rb

```diff
@@ -0,0 +1,28 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
+  attr_writer :login
+  validate :validate_username
+
+  def login
+    @login || self.username || self.email
+  end
```

Comment:
There's no need for `@login`. You can just create a publicmethod which does the similar as above

```ruby
  def login
    self.username || self.email
  end
```

Reply:
Done